### PR TITLE
[SYCL] Wait on queue in non-uniform-wk-gp-test

### PR DESCRIPTION
**Pull request by steffenlarsen**
Streams may leak if the associated queue is not synchronized. These changes forces an explicit wait on queue in non-uniform-wk-gp-test.

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
LGTM

**Comment by steffenlarsen** [comment]
Newest commit adds a failure case if exception was not thrown and reverses the default result for simplicity.

**Review by ghost — APPROVED** [review]

File: SYCL/Basic/diagnostics/non-uniform-wk-gp-test.cpp [hunk 1]
```diff
@@ -29,23 +30,23 @@ int test() {
...
       std::cout << "Test passed: caught the expected error." << std::endl;
-      return 0;
+      res = 0;
     } else {
       std::cout << E.what() << std::endl;
       std::cout << "Test failed: received error is incorrect." << std::endl;
-      return 1;
     }
   }
+  q.wait();
```
Comment (vladimirlaz) [inline SYCL/Basic/diagnostics/non-uniform-wk-gp-test.cpp:46]: Isn't it a bug that without queue::wait() the stream components are leaked on queue destruction?

Comment (steffenlarsen) [reply SYCL/Basic/diagnostics/non-uniform-wk-gp-test.cpp:46]: According to a comment in [`~Scheduler()`](https://github.com/intel/llvm/blob/sycl/sycl/source/detail/scheduler/scheduler.cpp#L367) it is expected behavior.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**